pkg/manager/impl/node_execution_manager.go

```diff
@@ -258,6 +272,21 @@ func (m *NodeExecutionManager) GetNodeExecution(
 		logger.Debugf(ctx, "failed to transform node execution model [%+v] to proto with err: %v", request.Id, err)
 		return nil, err
 	}
+	if nodeExecution.Closure.GetTaskNodeMetadata() != nil &&
+		nodeExecution.Closure.GetTaskNodeMetadata().DynamicWorkflow != nil &&
+		nodeExecution.Closure.GetTaskNodeMetadata().DynamicWorkflow.Id != nil {
+		dynamicWf, err := util.GetWorkflow(ctx, m.db, m.storageClient, *nodeExecution.Closure.GetTaskNodeMetadata().DynamicWorkflow.Id)
+		if err != nil {
+			logger.Warnf(ctx, "failed to fetch dynamic workflow definition for node [%+v] with err: %v", request.Id, err)
+		}
+		if dynamicWf.Closure != nil {
```

Comment:
So GetNodeExecution is only used by console (and clis) to visualize an execution, right? it's never called by propeller in any scenario... I'm just worried about the size of these messages being passed around if that's the case... and then I started thinking about these: https://developers.google.com/protocol-buffers/docs/reference/java/com/google/protobuf/FieldMask

Reply:
oh this is awesome! we should really integrate this throughout. would have been useful to have for the preliminary GetExecution performance hunt too

Reply:
and to answer your question, I don't see GetNodeExecution called anywhere in propeller

Reply:
I meant maybe propeller calls X that in turn calls GetNodeExecution? or something...

Reply:
well i checked plugins too, not sure what else there is to verify?

Reply:
we should never be calling GetNodeExecution in propeller. Question about the fieldmask, you cannot use that in a regular GetAPI IMO, as this changes the semantics of an API. The only way to do that is using a mask as an input, because then, this is the expectation by the user.

This is where GraphQL performs much better